dclab/rtdc_dataset/writer.py

```diff
@@ -210,14 +216,33 @@ def store_feature(self, feat, data, shape=None):
                         + 'in the `info["feature shapes"]` key of '
                         + "your plugin feature.")
                     shape = data.shape[1:]
-            if shape == data.shape:
-                data = data.reshape(1, *shape)
-            elif shape == data.shape[1:]:
-                pass
+            # Condition for single or multiple events to be saved
+            if shape == data.shape or shape == data.shape[1:]:
+                # Condition for ragged/contour data (array of arrays)
+                if data.ndim == 1 and isinstance(data[0], np.ndarray):
+                    self.write_ragged(group=events, name=feat, data=data)
+                # Condition for single image or array of shape (H, W). If the
+                # given data is a 2D, it is more likely a single image feature
```

Comment:
This comment is misleading. "more likely" than a ragged array?

Reply:
Sorry for the inconsistency with the commenting. It has to be under the below case (2D image).